grpc-gateway/util.lua

```diff
@@ -66,13 +67,44 @@ _M.map_message = function(field, default_values)
   end
 
   local request = {}
-  for name, _, field_type in pb.fields(field) do
+  for name, _, field_type,_,lbl in pb.fields(field) do
     if field_type:sub(1, 1) == "." then
-      sub, err = _M.map_message(field_type, default_values)
-      if err then
-        return nil, err
+      
+      -- This is a small workaround because the default_values begins as an empty table when a request comes into request.lua 
+      -- Its easier to work with the data, if the default_values was already populated with the inital JSON request. 
+      -- This could be moved completely outside the map_message function because we should only populate this once throughout the lifecycle of the request
+      -- Due to to the recursive calls, We check if the size of the table is empty or 0 in order to populate the data once.
+      if tablex.size(default_values) == 0 then  
```

Comment:
You might be able to use `table.isempty()` function which is extended on luajit2, bundled in OpenResty 1.15.8.1 https://github.com/openresty/luajit2#tableisempty and might be able to remove install and require `penlight`.

Reply:
Oh excellent I was unaware of that built in function, thanks.  I have removed the penlight dependency and I am making use of this instead..